src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -182,6 +182,23 @@ private function getEventFromTypeDeclaration(ContainerBuilder $container, string
 
         return $name;
     }
+
+    /**
+     * Resolve the listener method by checking if the class has only one public method.
+     * If such method exists, return its name, if multiple public methods exist, fallback on `__invoke`.
+     */
+    private function resolveListenerMethodFromPublicMethod(ContainerBuilder $container, string $id): string
+    {
+        if (
+            null === ($class = $container->getDefinition($id)->getClass())
+            || !($r = $container->getReflectionClass($class, false))
+            || 1 !== count($publicMethods = $r->getMethods(\ReflectionMethod::IS_PUBLIC))
+        ) {
+            return '__invoke';
```

Comment:
I kept this for BC, but I'm wondering if an exception should not be thrown here :thinking: